src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1650,22 +1662,44 @@ julia> flatten(df2, :b)
 │ 3   │ 2     │ r      │
 │ 4   │ 2     │ s      │
 
+julia> df3 = DataFrame(a = [1, 2], b = [[1, 2], [3, 4]], c = [[5, 6], [7]])
+2×3 DataFrame
+│ Row │ a     │ b      │ c      │
+│     │ Int64 │ Array… │ Array… │
+├─────┼───────┼────────┼────────┤
+│ 1   │ 1     │ [1, 2] │ [5, 6] │
+│ 2   │ 2     │ [3, 4] │ [7]    │
+
+julia> flatten(df3, [:b, :c])
+ERROR: ArgumentError: Vector lengths across columns in col, within the same row, must be the same
 ```
 """
-function flatten(df::AbstractDataFrame, col::ColumnIndex)
-    col_to_flatten = df[!, col]
-    lengths = length.(col_to_flatten)
-    new_df = similar(df[!, Not(col)], sum(lengths))
+function flatten(df::AbstractDataFrame,
+                 cols::Union{ColumnIndex, AbstractVector, Regex, Not, Between, All, Colon})
```

Comment:
since `flatten` is an expensive operation on all columns I think it is safer to add `_check_consistency(df)` as a first operation in this function as it will add only a little overhead, and we later heavily rely on the fact that the data frame is conistent.